src/vl-map-wmts-layer.js

```diff
@@ -0,0 +1,69 @@
+import {VlMapLayer} from '/src/vl-map-layer.js';
+import {OlWMTSSource, OlWMTSTileGrid, OlTileLayer, OlExtent} from '/node_modules/vl-mapactions/dist/vl-mapactions.js';
+/**
+ * VlMapWmtsLayer
+ *
+ * @class
+ * @classdesc Een WMTS (overlay) layer
+ *
+ * @extends VlMapBaseLayer
+ *
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/releases/latest|Release notes}
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-ui-map/issues|Issues}
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-map-wmts-layer.html|Demo}
+ */
+export class VlMapWmtsLayer extends VlMapLayer {
+  constructor() {
+    super();
+    this._layer = this._createWmtsLayer();
+  }
+
+  async connectedCallback() {
+    await super.connectedCallback();
+  }
+
+
+  get _projection() {
+    if (this.parentNode) {
+      return this.parentNode._projection;
+    }
+  }
+
+  get url() {
+    return this.getAttribute('data-vl-url');
+  }
+
+  _createWmtsLayer() {
+    const size = OlExtent.getWidth(this._projection.getExtent()) / 256;
+    const resolutions = new Array(16);
+    const matrixIds = new Array(16);
+    for (let z = 0; z < 16; ++z) {
+      resolutions[z] = size / Math.pow(2, z);
+      matrixIds[z] = z;
+    }
```

Comment:
Magic numbers misschien beter beschrijven

Reply:
Ik heb dat hier maar gewoon overgenomen van grb baselayer maar het was me niet duidelijk genoeg wat dat eigenlijk deed/doet moet ik toegeven. @Pieter-Beckers misschien wel?

Reply:
voelt wel magisch aan 🧙

Reply:
@Pieter-Beckers als jij hier meer over weet, wil jij dit anders even refactoren? Zo niet, moeten we misschien toch even dit stukje code begrijpen en wat leesbaarder maken.

Reply:
Goh ja, die getallen komen overeen met wat in de getCapabilities staat (zie TileMatrixSetLimits van BPL72VL in https://tile.informatievlaanderen.be/ws/raadpleegdiensten/wmts?request=getcapabilities ). Hardgecodeerd omdat we anders die xml moeten ophalen (wat niet altijd clientside gaat). Heel GRB specifiek hier dus.

Reply:
Ik zal het wat refactoren, uit elkaar trekken en op die manier duidelijk maken dat dat stukje GRB specifiek is.